determined-ai/determined
# fix: remember originally requested page on login [DET-6928]

## Description

issue:
- I don't get redirected to the page I was trying to visit after login


cause: we got multiple directions to the logout/login pages from a single logout and it ended up overriding the stored redirect path and eventually nulled out by a [prior change](https://github.com/determined-ai/determined/pull/3088)

solution:

- check for and strip redirect at the last step
- remove redundant redirects to logout/login
- avoid resetting authChecked on logout




## Test Plan

- scenario A
  - open two tabs
  - login on one the other should log in
  - on tab A move to /tasks
  - logout on tab B => tab A should log out (on visit + a small wait)
  - login on tab A or B => tab A should be directed to /tasks
- visiting /logout should log out every tab
- logout should clear auth tokens
- visit /tasks; logout by clicking the user avatar; login; you should be back on /tasks page





## Commentary (optional)

sequence of events for visiting a non root page when user is already logged out

before
<img width="526" alt="Screen Shot 2022-03-24 at 08 23 16" src="https://user-images.githubusercontent.com/12127420/159966017-3cdd1915-0dd0-4866-8b7e-9dcc5386208a.png">

after
<img width="1609" alt="Screen Shot 2022-03-24 at 08 24 49" src="https://user-images.githubusercontent.com/12127420/159966027-02e319cd-e1f6-4080-b08a-4d3b887c51d5.png">





## Checklist

See [Release Note](../docs/release-notes/README.md) for details.





[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
